Source/WebKit/UIProcess/Automation/WebAutomationSession.cpp

```diff
@@ -2307,7 +2307,24 @@ void WebAutomationSession::takeScreenshot(const Inspector::Protocol::Automation:
     bool scrollIntoViewIfNeeded = optionalScrollIntoViewIfNeeded ? *optionalScrollIntoViewIfNeeded : false;
     bool clipToViewport = optionalClipToViewport ? *optionalClipToViewport : false;
 
-#if PLATFORM(GTK)
+#if PLATFORM(COCOA)
+    // FIXME: <webkit.org/b/242215> We can currently only get viewport snapshots from the UIProcess, so fall back to
+    // taking a snapshot in the WebProcess if we need to snapshot a specific element. This has the side effect of not
+    // accurately showing all CSS transforms in the snapshot.
+    //
+    // There is still a tradeoff by going to the UIProcess for viewport snapshots on macOS. We can not currently get a
+    // snapshot of the entire viewport without the window's rounded corners being excluded. So we trade off those corner
+    // pixels for accurate pixels in the rest of the viewport which help us verify features like CSS transforms are
+    // actually behaving correctly.
+    if (!nodeHandle.isEmpty()) {
+        uint64_t callbackID = m_nextScreenshotCallbackID++;
+        m_screenshotCallbacks.set(callbackID, WTFMove(callback));
+
+        page->process().send(Messages::WebAutomationSessionProxy::TakeScreenshot(page->webPageID(), frameID, nodeHandle, scrollIntoViewIfNeeded, clipToViewport, callbackID), 0);
+        return;
+    }
+#endif
+
     Function<void(WebPageProxy&, std::optional<WebCore::IntRect>&&, Ref<TakeScreenshotCallback>&&)> takeViewSnapsot = [](WebPageProxy& page, std::optional<WebCore::IntRect>&& rect, Ref<TakeScreenshotCallback>&& callback) {
         page.callAfterNextPresentationUpdate([page = Ref { page }, rect = WTFMove(rect), callback = WTFMove(callback)](CallbackBase::Error error) mutable {
```

Comment:
I don't believe so, no. It's hard to draw a comparison between the two paths because the other path will actually paint fresh into a bitmap for us which means it may also not give us the "current frame" as visible and may be representative of the next frame.

The fact that GTK has been going down this path for a bit adds to my confidence.